mitodl/open-discussions
# Updated posts API to handle remove moderation

#### What are the relevant tickets?
Fixes #195 
Fixes #333 

#### What's this PR do?
Adds a flag for removing posts for moderators.

#### How should this be manually tested?

- Create a post as a normal contributor on a a channel (not a moderator).
- As a moderator go to `/api/v0/post/:id` and verify you see `"removed": false` in the response.
- `PATCH` with `{"removed": true}` and verify the response reflects the change
- Verify the post no longer shows up on the frontpage or channel listing
- You should still be able to view the post as it's content if you are a moderator or the OP, if you view as a third contributor, you should see `"text": "[removed]"`
- `PATCH` with `{"removed": false}` and verify the response reflects the change
- Verify the post shows up on the frontpage or channel listing again
